test/powershell/Language/Scripting/Scripting.Followup.Tests.ps1

```diff
@@ -146,4 +124,27 @@ public class NullStringTest {
         $result = & $powershell -noprofile -c '[System.Text.Encoding]::GetEncoding("IBM437").WebName'
         $result | Should -BeExactly "ibm437"
     }
+
+    It 'Return statement on the right side of an assignment should write the retrun value to outer pipe' {
+        function TestFunc1 {
+            ## The return value are not assigned to the variable but should be written to the outer pipe.
+            $Global:mylhsvar = if ($true) { return "one" }
```

Comment:
It might be worth adding a few more tests here as using `return` changes the behavior significantly